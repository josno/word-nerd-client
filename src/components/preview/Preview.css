.preview-words {
	font-size: 6rem;
	font-weight: bold;
	list-style: none;
}

.preview-words:hover {
	color: red;
	cursor: pointer;
}
