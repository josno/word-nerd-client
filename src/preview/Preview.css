.preview-box {
	height: 60vh;
}

.preview-words {
	font-size: 6rem;
	font-weight: bold;
}

.preview-words:hover {
	color: red;
	cursor: pointer;
}
