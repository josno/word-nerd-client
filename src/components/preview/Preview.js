import React from 'react';
import './Preview.css';

export default function Preview(props) {
	return (
		<div className="preview-box">
			<li className="preview-words">{props.word}</li>
		</div>
	);
}
