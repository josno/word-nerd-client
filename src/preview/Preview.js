import React, { Component } from 'react';
import './Preview.css';

class Preview extends Component {
	constructor(props) {
		super(props);
		this.state = {
			pageNumber: 1,
			renderedWord: ''
		};
	}

	componentDidMount() {
		//information here
	}

	render() {
		const { word } = this.props;
		const formattedWords = word.map((word, index) => (
			<p className="preview-words">
				<li key={index}>{word}</li>
			</p>
		));
		return <div className="preview-box">{formattedWords}</div>;
	}
}

export default Preview;
