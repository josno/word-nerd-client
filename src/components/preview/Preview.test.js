import React from './node_modules/react';
import Enzyme from './node_modules/enzyme';
import { BrowserRouter } from './node_modules/react-router-dom';
import { shallow } from './node_modules/enzyme';
import Adapter from './node_modules/enzyme-adapter-react-16';
import renderer from './node_modules/react-test-renderer';
import Preview from './Preview';
import toJson from './node_modules/enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('<Preview/>', () => {
	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<Preview />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders nothing when no props given', () => {
		const wrapper = shallow(<Preview />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('renders with props', () => {
		const tree = renderer.create(<Preview id={3} word={'red'} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
