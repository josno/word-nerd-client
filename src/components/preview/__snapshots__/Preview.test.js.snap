// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preview/> renders as expected 1`] = `
<div
  className="preview-box"
>
  <li
    className="preview-words"
  />
</div>
`;

exports[`<Preview/> renders nothing when no props given 1`] = `
<div
  className="preview-box"
>
  <li
    className="preview-words"
  />
</div>
`;

exports[`<Preview/> renders with props 1`] = `
<div
  className="preview-box"
>
  <li
    className="preview-words"
  >
    red
  </li>
</div>
`;
